src/editor/editor.js

```diff
@@ -0,0 +1,561 @@
+    const config = ev.detail.value || '';
+    if (!this._config || !this.hass) {
+      return;
+    }
+    // this._config = { config, ...this._entities };
+    fireEvent(this, 'config-changed', { config });
+  }
+
+  entitiesChanged(ev) {
+    ev.stopPropagation();
+    if (!this._config || !this.hass) {
+      return;
+    }
+    fireEvent(this, 'config-changed', { config: { ...this._config, entities: ev.detail } });
+  }
+
+  computeLabel(schema) {
+    const localized = this.hass.localize(`ui.panel.lovelace.editor.card.generic.${schema.name}`);
+    if (localized === '') {
+      return localize(`editor.form.card.${schema.name}`, this.hass);
+    } else {
+      return localized;
+    }
+  }
+
+  render() {
+    if (!this.hass || !this._config) {
+      return html``;
+    }
+
+
+    if (this.subElementEditorConfig !== undefined) {
+      return this.renderSubElement();
+      // return html`
+      //   <mini-graph-card-entity-editor
+      //     .hass=${this.hass}
+      //     .config=${this._entities[this.subElementEditorConfig.index]}
+      //     @go-back=${this.goBack}
+      //     @config-changed=${this.subElementChanged}
+      //   >
+      //   </mini-graph-card-entity-editor>
+      // `;
+    }
+
+    const data = {
+      hours_to_show: 24,
+      points_per_hour: 0.5,
+      aggregate_func: 'avg',
+      group_by: 'interval',
+      bar_spacing: 4,
+      line_width: 5,
+      color_thresholds_transition: 'smooth',
+      align_header: 'default',
+      align_icon: 'right',
+      align_state: 'left',
+      smoothing: true,
+      ...this._config,
+      show: { ...DEFAULT_SHOW, ...this._config.show },
+    };
+
+    return html`
+    <div>
+      <mini-graph-card-entities-editor
+        .hass=${this.hass}
+        .entities=${this._entities}
+        @config-changed=${this.entitiesChanged}
+        @edit-row=${this.editRow}
+      ></mini-graph-card-entities-editor>
+      <ha-form
+        .hass=${this.hass}
+        .data=${data}
+        .schema=${SCHEMA}
+        .computeLabel=${this.computeLabel}
+        @value-changed=${this.valueChanged}
+      ></ha-form>
+      <mini-graph-card-subpage-link
+        .name=${'Color thresholds'}
+        .icon=${mdiFormatColorFill}
+        @click=${this.editColorThresholds}
+      ></mini-graph-card-subpage-link>
+      <mini-graph-card-subpage-link
+        .name=${'State map'}
+        .icon=${mdiStateMachine}
+        @click=${this.editStateMap}
+      ></mini-graph-card-subpage-link>
+    </div>
+    `;
+  }
+
+  goBack() {
+    this.subElementEditorConfig = undefined;
+  }
+
+  editColorThresholds(ev) {
+    ev.stopPropagation();
+    if (!this._config || !this.hass) {
+      return;
+    }
+    this.subElementEditorConfig = { type: 'thresholds', index: 0 };
+  }
+
+  editStateMap(ev) {
+    ev.stopPropagation();
+    if (!this._config || !this.hass) {
+      return;
+    }
+    this.subElementEditorConfig = { type: 'statemap', index: 0 };
+  }
+
+  renderSubElement() {
```

Comment:
Thanks for the suggestion, maybe instead of subpages, `color_thresholds` and `state_map` whould be in their respective expansion panel under the appearance

Reply:
I did feel a bit unfamiliar with the concept of subpages, I must admit. However, I don't think, the entities' option can easily be folded into some expansion panel. So, we'd have subpages anyways, I guess. Then, why not use them consequently? Unless we find a way to also cascade the entities' options in another way...

My point here was in a slightly different direction: as both state map and colour thresholds create a list of schema-defined entries, this functionality could maybe be generalized and reused twice. Instead of the code being repeated (with some adaptations) twice.

Reply:
Yeah, the entities subpage is here to stay, but I thought about changing `color_thresholds` and `state_map` to use the custom `ha-form` element so it can be used in the main schema like this:
```js
 {
"the rest of the schema"
 },
 {
   name: 'color_thresholds',
   type: 'expandable',
   iconPath: mdiFormatColorFill,
   schema: [
     {
       type: 'mgc-list',
       schema: [
         {
           name: '',
           type: 'grid',
           column_min_width: '100px',
           schema: [
             {
               name: 'value',
               selector: { number: { step: 0.1 } },
             },
             {
               name: 'color',
               selector: { hex_color: {} },
             },
           ],
         },
       ],
     },
   ],
 },
```
That results in this:
![editor](https://github.com/user-attachments/assets/2f2bd882-3aba-4b4e-afb2-69be921cd941)

Reply:
Thanks for clarifying. That was exactly my suggestion as well. :-)